lib/jekyll_pages_api/page.rb

```diff
@@ -3,14 +3,17 @@
 module JekyllPagesApi
   # wrapper for a Jekyll::Page
   class Page
+    HTML_EXTS = %w(.html .md .markdown .textile)
     attr_reader :page
 
-    def initialize(page)
+    def initialize(page, site)
       @page = page
+      @site = site
     end
 
     def html?
-      %w(.html .md).include?(self.page.ext)
+      path = @page.path
```

Comment:
That's a very compelling argument. Had no idea about that `nil` behavior. You've probably just added a few hours or a couple of days back to my life, at least.